### PR TITLE
(lldb) Correctly fix a usage of `PATH_MAX`, and fix unit tests

### PR DESCRIPTION
**Pull request by royitaqi**
# Part 1: Correctly fix a usage of `PATH_MAX`

TL;DR: Adding a typedef `lldb_private::PathSmallString` which contains a hardcoded initial size (128).

https://github.com/llvm/llvm-project/pull/102223 broke Windows build by using `PATH_MAX` without defining it.

https://github.com/llvm/llvm-project/pull/104493 fixed it by including a header file which defines it.

However, **in general, we want to avoid using `PATH_MAX`**. While the above fixed the build (which is good), after a closer look at the code, we realized that we don't even need to use `PATH_MAX`. This is because we are using `SmallString<>`, which uses the parameter as a *initial* size, not *max* size. So here we can just put a reasonable small size (i.e. 1024).

# Part 2: Fix unit tests

After https://github.com/llvm/llvm-project/pull/104493 fixed the build break for Windows, unit test failure showed up for Windows. The root-cause is that the `FileSpec`'s in the unit tests are not style-specific. The fix is to apply either `WindowsSpec` or `PosixSpec` specifically.

### CODE REVIEW

File: lldb/source/Utility/RealpathPrefixes.cpp [hunk 1]
```diff
@@ -54,7 +53,7 @@ RealpathPrefixes::ResolveSymlinks(const FileSpec &file_spec) {
       LLDB_LOGF(log, "Realpath'ing support file %s", file_spec_path.c_str());
 
       // One prefix matched. Try to realpath.
-      llvm::SmallString<PATH_MAX> buff;
+      llvm::SmallString<1024> buff;
```
Comment (JDevlieghere) [inline lldb/source/Utility/RealpathPrefixes.cpp:56]: Should we give this a smaller initial value? Looks like most `SmallString`s in LLDB use either 64, 128 or 256. Maybe we should unify this and have a `lldb::PathSmallString` that typedefs to a reasonable value.

Comment (royitaqi) [reply lldb/source/Utility/RealpathPrefixes.cpp:56]: It's a good callout for code to share the same small initial value by defining a common constant.

But I think it's your call as to what value this should be (64, 128 or 256?). I can add the constant and use it here once you decide.

I can search the code base and see what is the most common value used and update the patch with this value as a recommendation.

File: lldb/include/lldb/lldb-private-types.h [hunk 2]
```diff
@@ -26,6 +27,8 @@ class Platform;
 class ExecutionContext;
 class RegisterFlags;
 
+typedef llvm::SmallString<128> PathSmallString;
```
Comment (clayborg) [inline lldb/include/lldb/lldb-private-types.h:30]: I would do 256 as a starting point for paths.

**Review by clayborg — APPROVED** [review]

**Comment by royitaqi** [comment]
Two changes made:
1. Add typedef `lldb_private::PathSmallString` with initial size 256. Use it in `RealpathPrefixes.cpp`.
2. Fix test errors in Windows (caused by by not specify WindowsSpec vs. PosixSpec).
